api/v2/performanceprofile_types.go

```diff
@@ -123,6 +126,36 @@ type NUMA struct {
 	TopologyPolicy *string `json:"topologyPolicy,omitempty"`
 }
 
+// Net defines a set of network related features
+type Net struct {
+	// UserLevelNetworking defines if the specified network devices . Defaults to "false".
+	UserLevelNetworking *bool `json:"userLevelNetworking,omitempty"`
+	// Devices contains a list of network device representations that will be
+	// set with a netqueue count equals to CPU.Reserved .
+	// If no devices are specified then the default is all devices.
+	Devices []Device `json:"devices,omitempty"`
+}
+
+// Device defines a way to represent a network device in several options:
+// device name, vendor ID, model ID, PCI path and MAC address
+type Device struct {
+	// Network device name
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Network device vendor ID represnted as a 16 bit Hexmadecimal number.
+	// +optional
+	VendorID *string `json:"vendor,omitempty"`
```

Comment:
On which cases does make it sense to specify only the vendor...

Reply:
In general all fields under device are acceptable fields by tuned net plugin - there will always be a cost for validation vs generalization.
If you specify only vendor you can cover a case where you want all devices of a specific vendor to be set with the downsized queue count for example.

Reply:
ok, makes sense

Reply:
I personally don't like this approach, users will want to use the same devices in all the nodes, otherwise they may not have the same hw, for example the same SRIOV VFs count. The way I understand the feature is that we need ways to identify the same Net devices. Is it customary use *the pair* VendorId:DeviceId to identify PCI devices.
We can use PPC to verify the assumptions, but it seems risky. I am OK with whatever you decide.